docs/environment-and-software/vscode.md

```diff
@@ -0,0 +1,197 @@
+
+### Interactive mode
+
+VS Code can use the Python Interactive mode to launch Jupyter notebooks, Python scripts, or debug code with the proper extensions.  
+
+The *Python Interpreter* is used to set the environment for interactive mode.  To select an environment, use the *Python: Select Interpreter* command from the Command Palette (⇧⌘P).  A valid environment is required to properly run Jupyter notebooks within VS Code and is a common use case for NCAR users.
+
+If you are connected to Derecho or Casper and have a Jupyter notebook or Python file preview open, you can click *Select Kernel* to view your available Python environments.  The list includes NCAR conda environments like `npl` and your own custom environments.
+
+![Kernel Select](vscode/media/selectkernel.png)
+
+Open a Jupyter notebook or Python file using either the terminal command `code` or with the File Explorer.  This will put you in the default kernel.  Like Jupyterhub, you will need to specify which kernel to use when executing cells in the notebook.  Not specifying a kernel will use your default kernel and could lead to import failures or unexpected behavior.
+
+!!! info 
+    Adding your conda-envs folder to the list of *Trusted* kernels will allow you to run notebooks with these kernels without being prompted to trust the kernel each time you run select a new kernel.
+
+## Jupyter Notebooks
+You can run cells similarly to Jupyterhub with the 'Run All' or an individual cell by hitting *Shift+Enter*.  Jupyter notebooks will use the VS Code kernel as your Python environment.
+
+![Jupyter Notebook](vscode/media/jnbkernel.png)
+
+Be mindful of the location of launching your notebooks.  If you use the File Explorer tab then you will be running on the login nodes.  
+Executing the command *code* from an interactive job will execute on the compute node.
+
+## Connecting to Compute Nodes
+The integration of PBS Batch jobs with VS Code requires some additional configuration for Derecho.  Casper requires that you launch your PBS job and then SSH directly into the compute node.  Derecho does not allow external connections to the compute nodes and requires a proxy node to connect to compute nodes from a VS Code Remote-SSH window.
+
+!!! example "Connecting to NCAR compute nodes"
+    === "Derecho"
+        Create a proxy option for Derecho login nodes to the compute nodes.  Add the following lines to your local SSH configuration file:
+
+        ```
+        Host dec????
+        HostName %h
+        ProxyCommand ssh -W %h:%p derecho
+
+        Host deg????
+        HostName %h
+        ProxyCommand ssh -W %h:%p derecho
+        ```
+
+        Once the proxy connection can be established with your edited local SSH configuration file, use the following steps to connect to a compute node:
+
+        1. Connect to a Derecho login node using Remote-SSH
+        2. From the Terminal, launch an interactive job using `qsub -I`
```

Comment:
A minor comment : I would add or `qinteractive` at the end:
From the Terminal, launch an interactive job using `qsub -I` or `qinteractive`